src/binary/ibc-setup/commands/ics20.spec.ts

```diff
@@ -0,0 +1,138 @@
+import fs from 'fs';
+import path from 'path';
+
+import { assert } from '@cosmjs/utils';
+import test from 'ava';
+import sinon from 'sinon';
+
+import { appFile } from '../../constants';
+
+import { createClient, Options, run } from './ics20';
+
+const fsWriteFileSync = sinon.stub(fs, 'writeFileSync');
+const fsReadFileSync = sinon.stub(fs, 'readFileSync');
+const consoleLog = sinon.stub(console, 'log');
+
+const mnemonic =
+  'enlist hip relief stomach skate base shallow young switch frequent cry park';
+
+const registryYaml = `
+version: 1
+
+chains:
+  musselnet:
+    chain_id: musselnet-4
+    # bech32 prefix for addresses
+    prefix: wasm
+    # this determines the gas payments we make (and defines the fee token)
+    gas_price: 0.025umayo
+    # the path we use to derive the private key from the mnemonic
+    hd_path: m/44'/108'/0'/1'
+    # you can include multiple RPC endpoints and it will rotate through them if
+    # one is down
+    rpc:
+      - https://rpc.musselnet.cosmwasm.com:443
+  local_wasm:
+    chain_id: testing
+    prefix: wasm
+    gas_price: 0.025ucosm
+    hd_path: m/44'/108'/0'/2'
+    rpc:
+      - http://localhost:26659
+  local_simapp:
+    chain_id: simd-testing
+    prefix: cosmos
+    gas_price: 0.025ucosm
+    hd_path: m/44'/108'/0'/3'
+    rpc:
+      - http://localhost:26658`;
+
+const app = {
+  src: 'local_wasm',
+  dest: 'local_simapp',
+};
+
+test.beforeEach(() => {
+  sinon.reset();
+});
+
+test.serial('ics20 create channels with new connection', async (t) => {
+  const ibcClientWasm = await createClient(mnemonic, {
+    prefix: 'wasm',
+    // hd_path: `m/44'/108'/0'/2'`,
```

Comment:
hd path is totally optional for these tests. nice to have to ensure addresses are different on both sides, but it doesn't really matter what key we use there

(You can just make a follow up issue later if this is a pain)

Reply:
I made `hd_path` optional in the `registry.yaml` so we can use defaults in tests.

Also: https://github.com/confio/ts-relayer/pull/94/files/1269e7e9a3a0090d7431e0d115e947af07b59aa4..d53fb34f8cb1d82a7fea5e9254b7af39c7521fb1#diff-58673c679ff81312a0f7389a4cf697deac398637a352094bd9d4832a5cf28ee4R49